Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -439,16 +446,23 @@ protected virtual GrpcChannelOptions GetChannelOptions()
 
         /// <summary>
         /// Returns whether or not a channel pool can be used if a channel is required. The default behavior is to return
-        /// true if and only if no quota project, scopes, credentials or token access method have been specified. Derived classes should
-        /// override this property if there are other reasons why the channel pool should not be used.
+        /// true if and only if no quota project, scopes, credentials or token access method have been specified and 
+        /// if UseJwtAccessWithScopes flag matches the flag in ChannelPool. 
+        /// Derived classes should override this property if there are other reasons why the channel pool should not be used.
         /// </summary>
         protected virtual bool CanUseChannelPool =>
             ChannelCredentials == null &&
             CredentialsPath == null &&
             JsonCredentials == null &&
             TokenAccessMethod == null &&
             Scopes == null &&
-            QuotaProject == null;
+            QuotaProject == null &&
+            UseJwtAccessWithScopes == GetChannelPool().UseJwtAccessWithScopes;
+
+        /// <summary>
+        ///  Returns whether or not self-signed JWTs will be used over OAuth tokens when OAuth scopes are explicitly set.
+        /// </summary>
+        protected virtual bool UseJwtAccessWithScopes { get; }
```

Comment:
This will never be set to true at the moment unless it's overridden. I think we either need to add another constructor accepting the value, or give this a setter as well and make it non-virtual. The latter is probably simpler - and means if we ever want to make it public, derived classes can just declare a new public property that delegates to this one.

(I tend to avoid virtual automatically-implemented properties - if it's overridden, the field in the base class is still there but unused, which is a bit confusing.)

Reply:
Also, we need to copy the value in CopyCommonSettings - which is easier if it's a regular automatically-implemented property.